packages/dds/tree/src/util/multimap.ts

```diff
@@ -0,0 +1,18 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export class Multimap<K, V> extends Map<K, V[]> {
+    add(key: K, value: V): void {
+        this.get(key).push(value);
+    }
+
+    get(key: K): V[] {
```

Comment:
You can save a map look up in both cases, and avoid the need to the cast at the bottom by doing the `get` first, and checking for undefined instead of calling `has`.